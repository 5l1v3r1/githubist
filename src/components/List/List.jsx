// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import styles from './List.scss';

type Props = {
  children: Node,
  columns: 1 | 2 | 3,
};

const List = ({ children, columns }: Props) => {
  const classes = cx(styles.list, styles[`columns${columns}`]);

  return <div className={classes}>{children}</div>;
};

export default List;
