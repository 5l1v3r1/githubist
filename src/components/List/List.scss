@import '../../theme/theme';

.list {
  display: grid;
  grid-auto-rows: 1fr;
  grid-column-gap: map-get($spacers, large);
  grid-row-gap: map-get($spacers, large);
}

.columns1 {
  grid-template-columns: 1fr;
}

.columns2 {
  grid-template-columns: 1fr 1fr;
}

.columns3 {
  grid-template-columns: 1fr 1fr 1fr;
}
