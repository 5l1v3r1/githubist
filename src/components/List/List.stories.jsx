// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import List from './List';

storiesOf('List', module).add('default', () => (
  <List columns={3}>
    <div>Column 1</div>
    <div>Column 2</div>
    <div>Column 3</div>
  </List>
));
