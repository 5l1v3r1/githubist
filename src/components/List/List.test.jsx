// @flow

import React from 'react';
import { shallow } from 'enzyme';

import List from './List';

it('renders correctly', () => {
  const wrapper = shallow(
    <List columns={3}>
      <div>Column 1</div>
      <div>Column 2</div>
      <div>Column 3</div>
    </List>,
  );

  expect(wrapper).toMatchSnapshot();
});
