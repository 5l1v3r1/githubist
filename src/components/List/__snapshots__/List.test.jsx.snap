// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <div>
    Column 1
  </div>
  <div>
    Column 2
  </div>
  <div>
    Column 3
  </div>
</div>
`;
